src/client/java/minicraft/level/Chunk.java

```diff
@@ -0,0 +1,39 @@
+package minicraft.level;
+
+import minicraft.level.tile.Tile;
+import minicraft.level.tile.Tiles;
+
+public class Chunk {
```

Comment:
This should probably be made into a struct, and integrated into the ChunkManager class. I know this makes working with chunks slightly harder, but it causes data to be stored closer in memory, which speeds up any interaction.

See: https://stackoverflow.com/questions/19385853/understanding-cache-friendly-data-oriented-objects-and-handles

Reply:
That's a good point!

Had I made it in C or C++, I probably would have made it a struct by default. Java OOP energy is corrupting my brain...

Do you have recommendations or thoughts on how a struct-like data structure is best implemented in Java?

If memory structure is also being discussed, I would just throw out there the potential of sticking the (2-byte) short tile id and (2-byte) short data together as one (4-byte) "int tile" in the format 0xTTTTDDDD (with T being the Tile id, and D being the tile data) 
I would make this choice if I were working in C, but I'm not sure if this would be effective or even beneficial _at all_ in Java. But just putting the idea out there in case it has any benefits.

Reply:
Uncertain, I am myself more familiar with Rust and .NET when it comes to this sort of stuff.

I think the easiest method would be to make two array of arrays. One for tiles, and another for data. The outer array would contain each chunk, and the inner array would be the contents of the chunk. This skips the need for a struct completely, and provides simple lookup and editing.

The other approach would be a simple array of chunk structs, each conaining two arrays, tiles and data. This approach might be more user-friendly, but I am unsure how Java treats structs internally, and if this would have any benifit at all.

Reply:
> If memory structure is also being discussed, I would just throw out there the potential of sticking the (2-byte) short tile id and (2-byte) short data together as one (4-byte) "int tile" in the format 0xTTTTDDDD (with T being the Tile id, and D being the tile data) I would make this choice if I were working in C, but I'm not sure if this would be effective or even beneficial _at all_ in Java. But just putting the idea out there in case it has any benefits.

I have seen this been done other places in Minicraft+, and it would certainly have a benefit if the first option is to be chosen. I am unsure about the performance cost/benifit, but it should theoretically reduce the memory footprint to about half, which is significant.

Reply:
That's fair, I would be fine with either implementation honestly. But I do agree that Chunk can (even _should_) be dissolved into ChunkManager.

If the struct version can be done simple enough and easy enough to read, it would probably be the better solution just in terms of the source code.
But I would expect the 2d arrays to be the more 'efficient' solution, just harder to understand what's going on by just looking at the code.